src/Datadog.Trace/Configuration/TracerSettings.cs

```diff
@@ -132,13 +133,12 @@ public TracerSettings(IConfigurationSource source)
             GlobalTags = GlobalTags.Where(kvp => !string.IsNullOrWhiteSpace(kvp.Key) && !string.IsNullOrWhiteSpace(kvp.Value))
                                    .ToDictionary(kvp => kvp.Key.Trim(), kvp => kvp.Value.Trim());
 
-            HeaderTags = source?.GetDictionary(ConfigurationKeys.HeaderTags) ??
+            var inputHeaderTags = source?.GetDictionary(ConfigurationKeys.HeaderTags, allowOptionalMappings: true) ??
                          // default value (empty)
                          new ConcurrentDictionary<string, string>();
```

Comment:
nit: not much need for this to be a `ConcurrentDictionary<>`, any more seeing as we're immediately enumerating it below and loading into a standard `Dictionary<>`

Reply:
Good point, will change 😄

Reply:
Done in b3f9c8c9975f0bbf1ada23a363a69edf4e77ac8d